src/engine/client/textrender.h

```diff
@@ -0,0 +1,232 @@
+/* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
+/* If you are missing that file, acquire a complete release at teeworlds.com.                */
+#ifndef ENGINE_CLIENT_TEXTRENDER_H
+#define ENGINE_CLIENT_TEXTRENDER_H
+
+#include <base/tl/sorted_array.h>
+#include <base/vmath.h>
+#include <engine/textrender.h>
+
+// ft2 texture
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_STROKER_H
+
+enum
+{
+	MAX_FACES = 16,
+	MAX_CHARACTERS = 64,
+	TEXTURE_SIZE = 1024,
+	PAGE_COUNT = 2,
+};
+
+// TODO: use SDF or MSDF font instead of multiple font sizes
+static int aFontSizes[] = {8,9,10,11,12,13,14,15,16,17,18,19,20,36,64};
+#define NUM_FONT_SIZES (sizeof(aFontSizes)/sizeof(int))
+#define PAGE_SIZE (TEXTURE_SIZE/PAGE_COUNT)
+
+struct CGlyph
+{
+	int m_FontSizeIndex;
+	int m_ID;
+	int m_AtlasIndex;
+	int m_PageID;
+	FT_Face m_Face;
+
+	bool m_Rendered;
+	float m_Width;
+	float m_Height;
+	float m_BearingX;
+	float m_BearingY;
+	float m_AdvanceX;
+	float m_aUvs[4];
```

Comment:
m_aUvs could be named more descriptively.

Reply:
since it is just UV coordinates. how about `m_aUvCoords`

Reply:
If that's correct, go for it. I assumed it was something like a UV scale.